### PR TITLE
Add DLONG support for osqp and qdldl

### LINKED ISSUE #2135: [Bug]: Cannot add google/osqp-cpp because osqp isn't built with `DLONG`
### What happened?

[osqp](https://registry.bazel.build/modules/osqp) doesn't define flag `DLONG` while building and [explicitly encourages against doing so](https://github.com/bazelbuild/bazel-central-registry/blob/main/modules/osqp/0.6.3/patches/add_build_file.patch#L26-L28).

### Version

Development (host) and target OS/architectures: linux

Output of `bazel --version`: 7.0.2

Language(s) and/or frameworks involved: C++

### How to reproduce

Try to build [pull 2134](https://github.com/bazelbuild/bazel-central-registry/pull/2134) and run 
`bazel build @osqp-cpp//:osqp++`

Fails with error:
```shell
 error: cannot initialize a member subobject of type 'c_int *' (aka 'int *') with an rvalue of type 'osqp::c_int *' (aka 'long long *')
      const_cast(objective_matrix_upper_triangle.outerIndexPtr()),
```
for all const_cast usages.

### PR DESCRIPTION
**Pull request by udaya2899**
This PR:
- Adds DLONG flag to build OSQP with 64-bit integers to fix #2135 
- Removes `QDLDL_LONG=0` flag in `qdldl` and sets INT_MAX to long long
- Without this change, google/osqp-cpp cannot build on Bazel, and cannot be published to BCR

The maintainer @wep21 has commented some concerns about DLONG and its relation with `qdldl` too. Please let me know what else I can do to enable this.

Closes #2135

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @wep21, modules you maintain (osqp) have been updated in this PR. Please review the changes.

**Review by mering — CHANGES_REQUESTED** [review]
Do not change existing versions. Move your changes to version `0.6.3.bcr.1`.

**Comment by udaya2899** [comment]
> Do not change existing versions. Move your changes to version `0.6.3.bcr.1`.

Sorry. Changed now.

**Comment by wep21** [comment]
@udaya2899 Could you also change float type in qdldl to match the one in osqp?

**Comment by bazel-io** [comment]
Hello @wep21, modules you maintain (osqp, qdldl) have been updated in this PR. Please review the changes.

File: modules/qdldl/0.1.7.bcr.1/patches/add_build_file.patch
Comment (mering) [inline modules/qdldl/0.1.7.bcr.1/patches/add_build_file.patch:30]:
```suggestion
+        # Keep the `float` type sync'd with the build of @osqp_internal and
```

File: modules/qdldl/0.1.7.bcr.1/source.json
Comment (mering) [inline modules/qdldl/0.1.7.bcr.1/source.json:7]: FYI: If this file didn't change, you can reference this via `../../0.1.7/patches/X.patch`.

**Review by mering — COMMENTED** [review]
Please also update the PR title and description to include qdldl.

**Comment by udaya2899** [comment]
Requesting @bazelbuild/bcr-maintainers to add the skip github link check as both osqp/osqp and osqp/qldl have only tags but not releases per [this blog](https://blog.bazel.build/2023/02/15/github-archive-checksum.html)

**Comment by udaya2899** [comment]
@wep21 The PR is ready for review again. Can you check?

**Comment by udaya2899** [comment]
@wep21 friendly ping. This will help me unblock #2134. Can you please review?

**Review by wep21 — APPROVED** [review]
LGTM. Thanks.

**Review by bazel-io — APPROVED** [review]
Hello @bazelbuild/bcr-maintainers, all modules in this PR have been approved by their maintainers. Please take a final look to merge this PR.

**Pull request merged**
